app/src/main/java/com/mapbox/navigation/examples/replay/HistoryFilesViewController.kt

```diff
@@ -0,0 +1,135 @@
+package com.mapbox.navigation.examples.replay
+
+import android.content.Context
+import com.mapbox.navigation.core.history.MapboxHistoryReader
+import com.mapbox.navigation.examples.R
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import java.io.File
+import java.util.Collections
+
+private const val REPLAY_HISTORY_FILE_NAME = "replay_history_activity.json"
+
+class HistoryFilesViewController(
+    private val historyFileDirectory: String?,
+    private val lifecycleScope: CoroutineScope
+) {
+
+    private var viewAdapter: HistoryFileAdapter? = null
+    private val historyFilesRepository = HistoryFilesDirectory()
+    private val historyFilesApi = HistoryFilesClient()
+
+    fun attach(
+        context: Context,
+        viewAdapter: HistoryFileAdapter,
+        result: (MapboxHistoryReader?) -> Unit
+    ) {
+        this.viewAdapter = viewAdapter
+        viewAdapter.itemClicked = { historyFileItem ->
+            when (historyFileItem.dataSource) {
+                ReplayDataSource.RAW_RES_DIRECTORY -> {
+                    requestFromRawResources(context, historyFileItem, result)
+                }
+                ReplayDataSource.FILE_DIRECTORY -> {
+                    requestFromFileCache(historyFileItem, result)
+                }
+                ReplayDataSource.HTTP_SERVER -> {
+                    requestFromServer(context, historyFileItem, result)
+                }
+            }
+        }
+    }
+
+    fun requestHistoryFiles(context: Context, connectionCallback: (Boolean) -> Unit) {
+        requestHistory(context, connectionCallback)
+    }
+
+    private fun requestHistory(context: Context, connectionCallback: (Boolean) -> Unit): Job {
+        return lifecycleScope.launch {
+            val drives = historyFilesApi.requestHistory().toMutableList()
```

Comment:
let's also put to the drives list labels for each category, smth like (first that I found 🎸 ):
![image](https://user-images.githubusercontent.com/8738058/173814790-d672618b-847d-4c01-bf02-e8ca11851896.png)

Reply:
Added sections to the RecyclerView.